DataDog/libdatadog
# Mini Agent: add trace normalization, root span calculation, logging

# What does this PR do?

- Introduces a logger (env_logger)
- Integrates the trace-normalization lib
- Use data from a tracer's request headers & trace root span tags to populate tracer payload fields
- Compute + tag top level spans
- Unit tests

# Motivation

Add basic features into the mini-agent for our serverless mvp

# Additional Notes

Future PRs will include:
- Change flushing logic to poll for traces to flush every 3 seconds, instead of based on buffer size
- Send JSON response to endpoint requests
- Trace sampling

# How to test the change?

Added unit tests.
Tested manually with a modified version of dd-trace-js. See prev PR: https://github.com/DataDog/libdatadog/pull/124
